contrib/mpi-proxy-split/mpi-wrappers/mpi_collective_p2p.c

```diff
@@ -643,6 +643,24 @@ int MPI_Iexscan(const void* sendbuf, void* recvbuf, int count,
 }
 #endif  // #ifdef ADD_UNDEFINED
 
+int
+MPI_Alltoall_internal(const void *sendbuf, int sendcount,
```

Comment:
This commit is good for now.

But I'd like to do a follow-on PR that modifies the code here for MPI_Alltoall_internal so that it makes calls to MPI_Send/Recv in the upper half, before calling the lower half.  We don't want it to directly make a call to MPI_Alltoall in the lower half.  Our principle should be to never call the lower half with a _collective communication_ call.  Then if there's a bug, we can guarantee that it's about the point-to-point layer.

Reply:
Thanks for the comment.